### ISSUE TITLE
Generated recovery link doesn't bypass email stage

### ISSUE THREAD

**Issue by Netti93** [goauthentik/authentik#9781]
**Describe the bug**
A recovery link generated for a user by an admin does not bypass email stage. Therefore the user then has to enter an email address, wait for that mail and restart the recovery process with the included link.

**To Reproduce**
1. Open user list in admin UI
2. Generate a recovery link or send a recovery mail for any user
3. The link is generated / mail is sent successfully, navigate to the recovery url
4. Email stage is shown and waiting for user input of an email address

**Expected behavior**
The email stage should be bypassed and the password prompt stage should be shown.

**Version and Deployment (please complete the following information):**

-   authentik version: 2024.4.2
-   Deployment: docker

**Additional context**
Tokens are generated correctly and were still valid when the recovery link was used.
The flow works correctly when the user requests a recovery link via "forgot password" and navigates to the url in the received mail.

**Possibly related bugs**
- https://github.com/goauthentik/authentik/issues/9671
- https://github.com/goauthentik/authentik/issues/2749

**Comment by nwinkelstraeter** [comment]
@Netti93 It seems this it not something that happens automatically.  It has to be implemented manually with an expression policy that skips the identification stage if the flow is restored from a token.  Have a look at: https://github.com/goauthentik/authentik/issues/3297

**Comment by Netti93** [comment]
@nwinkelstraeter yes, that is correct, but I do have imported the example flow and it is correctly skipping the identification stage if the token was created by the email stage itself. In that case the identfier value in the authentik_core_token table is something like "ak-email-stage-default-recovery-email-[...]".
But it doesn't work with a token created by an admin via e.g. the "Create Recovery Link" button which has a different identifier value structure like e.g. "[...]-password-reset"

**Comment by joachimtingvold** [comment]
I'm experiencing this issue as well. Admin-generated recovery URLs has no effect. Regardless of browser, the link simply prompts for username/password, and does not skip the verification stage.

I also observe the same difference in the `identifier` column in the `authentik_core_token` table that @Netti93 explains above.

**Comment by joachimtingvold** [comment]
It seems that both ways to manually create the token suffers from the same issue.

Clicking "Create Recovery Link" and "Email recovery link" both produces a `[…]-password-reset` value in the `identifier` column in the `authentik_core_token` table. Both of which doesn't work.

When a user creates a token themselves, a `ak-email-stage-default-recovery-email-[…]` value is used.

**Comment by melizeche** [comment]
@Netti93 try to adding an expression policy to the recovery flow

`return not request.context.get('is_restored', False)`

<img width="2532" height="1336" alt="Image" src="https://github.com/user-attachments/assets/4cb6b828-b4e2-49ca-bbdc-00038d7d9f21" />

I haven’t done thorough testing yet, but so far, it appears to work as intended.

**Comment by joachimtingvold** [comment]
> `return not request.context.get('is_restored', False)`

Doesn't seem to work on 2025.10.3, at least not that I can get working....

Also not sure why the `needs_documentation` tag was added to this issue?

**Comment by noriakisana** [comment]
any updates to this issue if it is fixed?
sofar it seemd not working without any workaround.
